vpc/service/branch_enis.go

```diff
@@ -1404,3 +1406,64 @@ func (vpcService *vpcService) createBranchENI(ctx context.Context, tx *sql.Tx, s
 
 	return iface, nil
 }
+
+func (vpcService *vpcService) ListTrunkToBranchENIMappings(ctx context.Context, req *vpcapi.ListTrunkToBranchENIMappingsRequest) (*vpcapi.ListTrunkToBranchENIMappingsResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "listTrunkToBranchENIMappings")
+	defer span.End()
+
+	mappings := []*vpcapi.TrunkENIMapping{}
+
+	tx, err := vpcService.db.BeginTx(ctx, &sql.TxOptions{
+		ReadOnly:  true,
+		Isolation: sql.LevelRepeatableRead,
+	})
+	if err != nil {
+		err = errors.Wrap(err, "Cannot start database transaction")
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
```

Comment:
Generally the approach in the code is:
```
defer func() {
   _ = tx.Rollback()
}()

... Codes ...

err = tx.Commit()
if err != nil {
   tracehelper...
   return err
}
```

If you tx rollback a txn that's already been committed it doesn't cause a problem (throws an error saying not in txn).